server/util/timeutil/timeutil.go

```diff
@@ -5,9 +5,24 @@ import (
 	"time"
 )
 
-// FromMillis converts a Unix timestamp in milliseconds to a `time.Time`.
+// FromMillis converts a Unix timestamp in milliseconds to a Time.
 func FromMillis(timestampMillis int64) time.Time {
-	return time.Unix(0, timestampMillis*1e6)
+	return time.Unix(0, timestampMillis*int64(time.Millisecond))
+}
+
+// FromUsec converts a Unix timestamp in microseconds to a Time.
+func FromUsec(timestampUsec int64) time.Time {
+	return time.Unix(0, timestampUsec*int64(time.Microsecond))
+}
+
+// ToMillis converts a Time to a timestamp in milliseconds since the Unix epoch.
+func ToMillis(t time.Time) int64 {
+	return t.UnixNano() / int64(time.Millisecond)
+}
+
+// ToUsec converts a Time to a timestamp in microseconds since the Unix epoch.
+func ToUsec(t time.Time) int64 {
+	return t.UnixNano() / int64(time.Microsecond)
 }
```

Comment:
I am biased but I find these math statements involving predefined time.Duration values to be quite confusing! Could we just use the numbers instead in this function?

Since you're only doing this math in one place, we only have to check that it's correct once and we're good